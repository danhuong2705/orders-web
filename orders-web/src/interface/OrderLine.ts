export interface IOrderLine {
  sku: string;
  name: string;
  price: number;
  imageUrl: string;
  quantity: number;
}
export interface IProductCart extends IOrderLine {
  addedAt: Date;
}
