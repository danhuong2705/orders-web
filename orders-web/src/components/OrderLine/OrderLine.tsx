import styled from "@emotion/styled";
import React, { useState } from "react";
import { formatCurrency } from "../../helpers";
import { IOrderLine } from "../../interface/OrderLine";

interface OrderLineProps {
  sku: string;
  name: string;
  quantity: number;
  imageUrl: string;
  price: number;
  onIncreaseProduct?: (product: IOrderLine) => void;
  onDecreaseProduct?: (product: IOrderLine) => void;
}
const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 0.25rem;
  align-items: center;
`;
const Image = styled.img`
  width: 5rem;
  height: 5rem;
  object-fit: contain;
`;
const Name = styled.div`
  margin-left: 0.5rem;
`;
const QuantityBlock = styled.div`
  margin-right: 3rem;
  display: flex;
  justify-content: space-between;
`;
const IncreaseBtn = styled.button`
  cursor: pointer;
  background-color: #ffffff;
  border: 1px solid #e0e0e0;
  width: 2.5rem;
  height: 1.5rem;
  :hover {
    border: 1px solid #000000;
  }
`;
const Price = styled.div``;
const DecreaseBtn = styled(IncreaseBtn)``;
const Quantity = styled.div`
  margin: 0 1rem;
`;
const BlockRight = styled.div`
  display: flex;
`;
const OrderLine: React.FC<OrderLineProps> = ({
  sku,
  name,
  quantity,
  imageUrl,
  price,
  onDecreaseProduct,
  onIncreaseProduct,
}: OrderLineProps) => {
  const [quantityValue, setQuantityValue] = useState(quantity);

  return (
    <Wrapper>
      <Wrapper>
        <Image src={imageUrl} alt="product-image" />
        <Name>{name}</Name>
      </Wrapper>
      <BlockRight>
        <QuantityBlock>
          {onIncreaseProduct && (
            <IncreaseBtn
              onClick={() => {
                onIncreaseProduct({
                  sku,
                  name,
                  price,
                  imageUrl,
                  quantity: quantityValue,
                });
                setQuantityValue(quantityValue + 1);
              }}
            >
              +
            </IncreaseBtn>
          )}
          <Quantity>{quantityValue}</Quantity>
          {onDecreaseProduct && (
            <DecreaseBtn
              onClick={() => {
                onDecreaseProduct({
                  sku,
                  name,
                  price,
                  imageUrl,
                  quantity: quantityValue,
                });
                setQuantityValue(quantityValue - 1);
              }}
            >
              -
            </DecreaseBtn>
          )}
        </QuantityBlock>
        <Price>{formatCurrency(price)}</Price>
      </BlockRight>
    </Wrapper>
  );
};
export default OrderLine;
