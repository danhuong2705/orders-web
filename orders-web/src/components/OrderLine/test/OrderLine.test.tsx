import React from "react";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import OrderLine from "../OrderLine";

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe("<OrderLine />", () => {
  it("should render correctly", async () => {
    const { container } = render(
      <BrowserRouter>
        <OrderLine
           sku="21232"
           name="121323"
           quantity={1}
           imageUrl=""
           price={20000}
           onDecreaseProduct={() => {}}
           onIncreaseProduct={() => {}}
        />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
