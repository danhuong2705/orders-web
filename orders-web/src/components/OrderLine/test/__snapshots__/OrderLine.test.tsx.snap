// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItem /> should render correctly 1`] = `
<div
  class="css-1c13cqv"
>
  <div
    class="css-1c13cqv"
  >
    <img
      alt="product-image"
      class="css-vdzwhy"
      src=""
    />
    <div
      class="css-tubh1u"
    >
      121323
    </div>
  </div>
  <div
    class="css-ov1ktg"
  >
    <div
      class="css-wi4tla"
    >
      <button
        class="css-tg6web"
      >
        +
      </button>
      <div
        class="css-17qyic4"
      >
        1
      </div>
      <button
        class="css-tg6web"
      >
        -
      </button>
    </div>
    <div
      class="css-0"
    >
      20.000 ₫
    </div>
  </div>
</div>
`;

exports[`<OrderLine /> should render correctly 1`] = `
<div
  class="css-1c13cqv"
>
  <div
    class="css-1c13cqv"
  >
    <img
      alt="product-image"
      class="css-vdzwhy"
      src=""
    />
    <div
      class="css-tubh1u"
    >
      121323
    </div>
  </div>
  <div
    class="css-ov1ktg"
  >
    <div
      class="css-wi4tla"
    >
      <button
        class="css-tg6web"
      >
        +
      </button>
      <div
        class="css-17qyic4"
      >
        1
      </div>
      <button
        class="css-tg6web"
      >
        -
      </button>
    </div>
    <div
      class="css-0"
    >
      20.000 ₫
    </div>
  </div>
</div>
`;
